docs/source/rfc-configuration-overhaul.md

```diff
@@ -0,0 +1,721 @@
+
+### Proposal 6 - Exhaustive Error Messages
+
+#### Objective
+
+To enhance user troubleshooting by offering comprehensive and descriptive error messages during the configuration stage, specifically for two main types of configuration errors.
+
+#### Rationale
+
+Diagnosing and resolving configuration errors can be challenging without detailed insights. A more descriptive error messaging system offers the following benefits:
+
+1. **Parsing with Precision:**
+   - **Highlight Invalid Data's Location:** Pinpoint the exact location of errors, making it easier for users to address.
+   - **Identify Missing Fields:** Rapidly flag any necessary but absent data, speeding up the configuration process.
+   - **Spot Unknown Fields:** Alert users to any unrecognized fields, helping them to quickly catch and fix typographical mistakes or redundant information.
+2. **Clear Directions for Domain-Specific Errors:**
+   - **Highlighting Semantic Issues:** Inform users about errors or warnings tied to the higher-level domain details specific to Iroha. This ensures that users are aware of any nuances or complexities related to Iroha itself, allowing for a more refined and accurate configuration.
+3. **Eliminate Guesswork:**
+   - **Full Transparency in Reporting:** Ensure users are immediately and fully informed of any issues, preventing oversights and minimizing confusion.
+4. **Consistency in Communication:**
+   - **Uniform, Informative Messaging:** Maintain a consistent style across error messages, bolstering clarity and user understanding.
+5. **Streamline Multi-Error Solutions:**
+   - **Bulk Reporting of Missing Fields:** If several fields are missing, display them all simultaneously for more efficient troubleshooting.
+
+#### Potential Solutions & Samples
+
+To illustrate the potential of this system, a Proof-of-Concept (PoC) repository should be created, showcasing the clarity and utility of these enhanced error messages in Rust.
+
+#### Summary
+
+Implementing exhaustive error messages will drastically enhance user troubleshooting, ensuring a smoother and more efficient configuration process for Iroha users.
+
+#### See Also
+
+- [[suggestion] Enhance configuration parsing with precise field locations · Issue #3470 · hyperledger/iroha](https://github.com/hyperledger/iroha/issues/3470)
+
+### Proposal 7 - Trace Configuration Resolution
+
+#### Objective
+
+Implement a mechanism that displays the sequence in which configuration is determined. This allows users to grasp the sources from which configuration values are derived, whether they're being overridden, and which resort to default values.
+
+#### Rationale
+
+Transparency in configuration resolution is vital for troubleshooting and ensuring the correct setup. By providing clear feedback on how parameters are set, users can identify potential misconfigurations or conflicts quickly.
+
+#### Proposed Tracing Output
```

Comment:
Should probably include the resultant config values after all resolution stages.

Reply:
Addressed, changed the sample output.